File: frontend/src/lib/types/skill-usage.ts
/** Phase 3 스킬 축 usage 요약 — 실측 귀속만 (평가 런 토큰/비용 + 채팅 실행 횟수). */

export type SkillUsageDailyPoint = {
  readonly date: string
  readonly tokens_in: number
  readonly tokens_out: number
  readonly cost_usd: number
  readonly execution_count: number
}

export type SkillUsageSummary = {
  readonly skill_id: string
  readonly days: number
  readonly tokens_in: number
  readonly tokens_out: number
  /** 단가가 알려진 이벤트의 비용 합 — 단가 없는 이벤트는 기여하지 않는다. */
  readonly cost_usd: number
  readonly priced_event_count: number
  /** 토큰은 썼지만 단가가 없어 비용을 모르는 이벤트 수 (모름 ≠ 무료). */
  readonly unpriced_token_event_count: number
  readonly evaluation_run_count: number
  readonly chat_execution_count: number
  readonly daily: readonly SkillUsageDailyPoint[]
}

File: backend/app/schemas/skill_usage.py
"""Skill-axis usage summary schemas (Phase 3 spec §6)."""

from __future__ import annotations

import uuid

from pydantic import BaseModel, ConfigDict


class SkillUsageDailyPointResponse(BaseModel):
    model_config = ConfigDict(frozen=True)

    date: str
    tokens_in: int
    tokens_out: int
    cost_usd: float
    execution_count: int


class SkillUsageSummaryResponse(BaseModel):
    model_config = ConfigDict(frozen=True)

    skill_id: uuid.UUID
    days: int
    tokens_in: int
    tokens_out: int
    # Sum of *known* costs only — events without pricing contribute nothing.
    cost_usd: float
    priced_event_count: int
    # Events that consumed tokens but had no pricing (cost unknown ≠ free).
    unpriced_token_event_count: int
    evaluation_run_count: int
    chat_execution_count: int
    daily: list[SkillUsageDailyPointResponse]
